pkg/daemon/ceph/osd/volume.go

```diff
@@ -779,7 +779,7 @@ func (a *OsdAgent) initializeDevicesLVMMode(context *clusterd.Context, devices *
 			}
 
 			for _, report := range cvReports {
-				if report.BlockDB != mdPath {
+				if report.BlockDB != mdPath && !strings.HasSuffix(mdPath, report.BlockDB) {
```

Comment:
Would be nice to have case covered in the `TestInitializeBlockWithMD` unit test.

Reply:
Added a test for this in TestInitializeBlockWithMD (fails without PR applied, passes with PR applied).